src/runtime_src/core/tools/xbutil2/OO_AieClockFreq.cpp

```diff
@@ -0,0 +1,149 @@
+    ("help,h", po::bool_switch(&m_help), "Help to use this sub-command")
+  ;
+}
+
+void
+OO_AieClockFreq::execute(const SubCmdOptions& _options) const
+{
+
+  XBU::verbose("SubCommand option: clock");
+
+  XBU::verbose("Option(s):");
+  for (auto & aString : _options)
+    XBU::verbose(std::string(" ") + aString);
+
+  // Honor help option first
+  if (std::find(_options.begin(), _options.end(), "--help") != _options.end()) {
+    printHelp();
+    return;
+  }
+
+  // Parse sub-command ...
+  po::variables_map vm;
+
+  try {
+    po::store(po::command_line_parser(_options).options(m_optionsDescription).positional(m_positionalOptions).run(), vm);
+    po::notify(vm); // Can throw
+  }
+  catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << "\n\n";
+    printHelp();
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+
+  // Exit if neither action or device specified
+  if(m_help || m_device.empty()) {
+    printHelp();
+    return;
+  }
+
+  // Check if set/get is used
+  if(!m_get && m_freq.length() == 0) {
+    std::cerr << "ERROR: Neither `set` nor `get' is used" << std::endl;
+    std::cerr << "please use any one of set/get and rerun" << std::endl;
+    printHelp();
+    return;
+  }
+
+  // Check if partition_id is provided else print Warning!
+  if(!vm.count("partition"))
+      std::cout << "WARNING: `partition` option is not provided, using default partition id value '1'" << std::endl;
+
+  // Collect all of the devices of interest
+  std::set<std::string> deviceNames;
+  xrt_core::device_collection deviceCollection;
+  deviceNames.insert(boost::algorithm::to_lower_copy(m_device));
+  
+  try {
+    XBU::collect_devices(deviceNames, true /*inUserDomain*/, deviceCollection);
+  } catch (const std::runtime_error& e) {
+    // Catch only the exceptions that we have generated earlier
+    std::cerr << boost::format("ERROR: %s\n") % e.what();
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+
+  bool errorOccured = false;
+  for (auto& device : deviceCollection) {
+    try {
+        if(m_get) {
+            double freq_part = static_cast<double>(to_mega(xrt_core::device_query<qr::aie_get_freq>(device, m_partition_id)));
+            std::cout << boost::format("INFO: Frequency value of aie partition_id %d is : %lu MHz\n") % m_partition_id % freq_part ;
+        }
+        if(m_freq.length() > 0) {
+            uint64_t freq = 0;
+            try {
+		//convert freq to hertz(Hz)
+                freq = XBUtilities::string_to_bytes(m_freq);
+            }
+            catch(const xrt_core::error&) {
+                std::cerr << "Freq value provided with `set` option is invalid. Please specify proper units and rerun" << std::endl;
+                std::cerr << "eg: 'B', 'K', 'M', 'G' " << std::endl;
+                throw xrt_core::error(std::errc::operation_canceled);
+            }
+
+            xrt_core::device_query<qr::aie_set_freq_req>(device, m_partition_id, freq) ? 
```

Comment:
Hmm...it might be beneficial to the user if the following code was added:
1. Print out the clock frequency before the change.
2. Change the clock frequency.
3. Validate the clock frequency is running the the value being set.